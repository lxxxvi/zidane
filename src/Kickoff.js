import React from 'react';
import moment from 'moment';

class Kickoff extends React.Component {
  render() {
    return (
      <time
        dateTime={this.props.time}
      >
        {this.timeUntilKickoff()}
      </time>
    );
  }

  timeUntilKickoff() {
    const kickoff = moment(this.props.time);
    const now = moment();

    return moment.duration(kickoff.diff(now)).humanize(true);
  }
}

export default Kickoff;
