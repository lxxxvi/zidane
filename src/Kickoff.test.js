import React from 'react';
import renderer from 'react-test-renderer';
import MockDate from 'mockdate';
import moment from 'moment';
import Kickoff from './Kickoff';

afterEach(() => {
  MockDate.reset();
});

test("Kickoff", () => {
  const fakeTime = moment('2020-06-12 17:00:00');

  MockDate.set(fakeTime);

  const component = renderer.create(
    <Kickoff time="2020-06-12 18:00:00" />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
