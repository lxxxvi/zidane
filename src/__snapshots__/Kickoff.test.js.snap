// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kickoff 1`] = `
<time
  datetime="2020-06-12 18:00:00"
>
  in an hour
</time>
`;
